### PR TITLE
Sd 107 join create league

### PR DESCRIPTION
**Pull request by asnewman**
Does not include league management (e.g. storing the new league, redirecting to the page, etc...)

### CODE REVIEW

File: src/api/api.js [hunk 1]
```diff
@@ -2,27 +2,33 @@ import axios from 'axios';
...
+   }
+};
 
 const urls = {
-  charts: `${baseURL}/charts`,
-  transactions: `${baseURL}/transactions`,
+   charts: baseURL + '/charts',
+   transactions: baseURL + '/transactions',
+   portfolios: baseURL + '/portfolios',
+   leagues: baseURL + '/leagues'
```
Comment (sshaul) [inline src/api/api.js]: Why did you change it not to have the `${baseURL}` format?

File: src/api/api.js [hunk 2]
```diff
@@ -31,8 +37,26 @@ export const getStockHistory = async params => await axios.get(`${urls.charts}/$
  * action
  * portfolioId
  */
-export const createTransaction = async params => await axios.post(urls.transactions, params, getRequestConfig());
-
-export const registerUser = async (userInfo) => {
-  await axios.post('http://localhost:5005/api/v1.0/users', userInfo);
```
Comment (sshaul) [inline src/api/api.js:37]: I don't know if you want to get rid of the registerUser function here... Did you pull before you did this pr?

File: src/components/Button/Button.css [hunk 3]
```diff
@@ -11,15 +11,17 @@ body {
...
   font-family: 'Assistant';
   font-size: 20px;
   padding: 10px 0px;
   color: #f7f8f9;
   border-radius: 3px;
   cursor: pointer; }
 
+.Button button {
+  background-color: #3ee7ad;
+  border: none; }
```
Comment (sshaul) [inline src/components/Button/Button.css:23]: Why did you separate these into two css objects instead of the one?

File: src/components/Navigation/JoinCreateLeagueBtn/JoinCreateLeague.jsx [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+
+  openModal = () => {
+    this.setState({ isModalOpen: true });
+  };
+
+  closeModal = () => {
+    this.setState({ isModalOpen: false });
+  };
+
+  handleSelectChange = (selectedOption) => {
```
Comment (sshaul) [inline src/components/Navigation/JoinCreateLeagueBtn/JoinCreateLeague.jsx]: Might be nice to make this more specific (what's being selected?)

File: src/components/Navigation/JoinCreateLeagueBtn/JoinCreateLeague.jsx [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+    return this.state.name && 
+           this.state.leagueName && 
+           this.state.buyingPower && 
+           parseInt(this.state.buyingPower, 10) > 0 && 
+           this.validateDates(this.state.selectedStartDay, this.state.selectedEndDay);
+  }
+
+  validateDates = (start, end) => {    
+    return start &&
+           end &&
```
Comment (sshaul) [inline src/components/Navigation/JoinCreateLeagueBtn/JoinCreateLeague.jsx:99]: Not really important but didn't you already check if start and end are null in lines 76 and 77?

File: src/components/Navigation/JoinCreateLeagueBtn/JoinCreateLeague.jsx [hunk 6]
```diff
@@ -0,0 +1,222 @@
...
+  /**
+   * Makes the join league api call and then redirects to new league page
+   */
+  executeJoinLeague = async () => {
+    try {
+      await joinLeague({
+        inviteCode: this.state.inviteCode,
+        name: this.state.name
+      });
+      this.setState({isModalOpen: false}, () => this.props.alert.show('League joined.'));
```
Comment (sshaul) [inline src/components/Navigation/JoinCreateLeagueBtn/JoinCreateLeague.jsx:113]: Where does it redirect to new league page?

**Review by sshaul — CHANGES_REQUESTED** [review]

File: src/api/api.js (hunk 1 above)
Comment (asnewman) [reply src/api/api.js]: not sure, will fix

File: src/api/api.js (hunk 2 above)
Comment (asnewman) [reply src/api/api.js:37]: hmmm that's not good...

File: src/components/Button/Button.css (hunk 3 above)
Comment (asnewman) [reply src/components/Button/Button.css:23]: Doh!

File: src/components/Navigation/JoinCreateLeagueBtn/JoinCreateLeague.jsx (hunk 4 above)
Comment (asnewman) [reply src/components/Navigation/JoinCreateLeagueBtn/JoinCreateLeague.jsx]: It's super long now!

File: src/components/Navigation/JoinCreateLeagueBtn/JoinCreateLeague.jsx (hunk 5 above)
Comment (asnewman) [reply src/components/Navigation/JoinCreateLeagueBtn/JoinCreateLeague.jsx:99]: never mind

Comment (asnewman) [reply src/components/Navigation/JoinCreateLeagueBtn/JoinCreateLeague.jsx:99]: i fixed it

File: src/components/Navigation/JoinCreateLeagueBtn/JoinCreateLeague.jsx (hunk 6 above)
Comment (asnewman) [reply src/components/Navigation/JoinCreateLeagueBtn/JoinCreateLeague.jsx:113]: Functionality won't be added until login page is done because i don't have token yet

Comment (sshaul) [reply src/components/Navigation/JoinCreateLeagueBtn/JoinCreateLeague.jsx:113]: 👍

**Review by sshaul — APPROVED** [review]
